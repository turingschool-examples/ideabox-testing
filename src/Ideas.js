import React from 'react';
import Card from './Card';
import './Ideas.css';

const Ideas = ({ideas, deleteIdea}) => {

  const ideaCards = ideas.map(idea => {
    return (
      <Card
        title={idea.title}
        description={idea.description}
        id={idea.id}
        key={idea.id}
        deleteIdea={deleteIdea}
      />
    )
  })

  return (
    <div className='ideas-container'>
      {ideaCards}
    </div>
  )
}

export default Ideas;
