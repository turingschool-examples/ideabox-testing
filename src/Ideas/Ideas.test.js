import React from 'react';
import { shallow } from 'enzyme';
import Ideas from './Ideas';

describe('Ideas', () => {
  it('should match snapshot', () => {
    const mockIdeas = [
      {id: 1, title: 'Sweaters for pugs', description: 'To keep them warm'},
      {id: 2, title: 'A romcom', description: 'But make it ghosts'},
      {id: 3, title: 'A game show called Ether/Or', description: 'When you lose you get chloroformed'},
    ];

    const wrapper = shallow(<Ideas 
      ideas={mockIdeas} 
      removeIdea={jest.fn()} 
    />);

    expect(wrapper).toMatchSnapshot();
  });
});