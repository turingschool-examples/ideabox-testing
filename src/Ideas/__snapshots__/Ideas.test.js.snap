// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ideas should match snapshot 1`] = `
<Fragment>
  <h2>
    Ideas Component
  </h2>
  <section
    className="ideas"
  >
    <Card
      description="To keep them warm"
      id={1}
      isFavorite={true}
      key="1"
      removeIdea={[MockFunction]}
      title="Sweaters for pugs"
    />
    <Card
      description="But make it ghosts"
      id={2}
      isFavorite={true}
      key="2"
      removeIdea={[MockFunction]}
      title="A romcom"
    />
    <Card
      description="When you lose you get chloroformed"
      id={3}
      isFavorite={true}
      key="3"
      removeIdea={[MockFunction]}
      title="A game show called Ether/Or"
    />
  </section>
</Fragment>
`;
